src/Client.php

```diff
@@ -106,39 +113,54 @@ public function withBearerToken($bearerToken)
         $this->auth = [$bearerToken];
         return $this;
     }
-	
+
+    /**
+     * Configure the client to either catch web request exceptions and return a message or throw the exception
+     *
+     * @param bool $catchExceptions
+     * @return AvaTaxClient
+     */
+    public function setCatchExceptions($catchExceptions = true)
+    {
+        $this->catchExceptions = $catchExceptions;
+        return $this;
+    }
+
     /**
      * Make a single REST call to the AvaTax v2 API server
      *
-     * @param string $apiUrl           The relative path of the API on the server
-     * @param string $verb             The HTTP verb being used in this request
-     * @param string $guzzleParams     The Guzzle parameters for this request, including query string and body parameters
+     * @param string $apiUrl The relative path of the API on the server
+     * @param string $verb The HTTP verb being used in this request
+     * @param string $guzzleParams The Guzzle parameters for this request, including query string and body parameters
+     * @return mixed|string
      */
     protected function restCall($apiUrl, $verb, $guzzleParams)
     {
         // Set authentication on the parameters
-        if(count($this->auth) == 2){
-		if (!isset($guzzleParams['auth'])){
-			$guzzleParams['auth'] = $this->auth;
-		}
-		$guzzleParams['headers'] = [
-			'Accept' => 'application/json',
-			'X-Avalara-Client' => "{$this->appName}; {$this->appVersion}; PhpRestClient; 17.5.0-67; {$this->machineName}"
-		];
-	 } else {
-		$guzzleParams['headers'] = [
-			'Accept' => 'application/json',
-			'Authorization' => 'Bearer '.$this->auth[0],
-			'X-Avalara-Client' => "{$this->appName}; {$this->appVersion}; PhpRestClient; 17.5.0-67; {$this->machineName}"
-		];
-	 }
-
-        // Contact the server
+        if (count($this->auth) == 2) {
+            if (!isset($guzzleParams['auth'])) {
+                $guzzleParams['auth'] = $this->auth;
+            }
+            $guzzleParams['headers'] = [
+                'Accept' => 'application/json',
+                'X-Avalara-Client' => "{$this->appName}; {$this->appVersion}; PhpRestClient; 17.5.0-67; {$this->machineName}"
+            ];
+        } else {
+            $guzzleParams['headers'] = [
+                'Accept' => 'application/json',
+                'Authorization' => 'Bearer '.$this->auth[0],
+                'X-Avalara-Client' => "{$this->appName}; {$this->appVersion}; PhpRestClient; 17.5.0-67; {$this->machineName}"
+            ];
+        }
+
         try {
             $response = $this->client->request($verb, $apiUrl, $guzzleParams);
             $body = $response->getBody();
             return json_decode($body);
         } catch (\Exception $e) {
+            if ($this->catchExceptions) {
```

Comment:
I think this is the opposite of what you want.

Reply:
This has been fixed.